math/mathcore/inc/TMath.h

```diff
@@ -20,6 +20,9 @@
 #include <limits>
 #include <cmath>
 
+#include "Math/Types.h"
+#include <VecCore/VecCore>
+#include <Vc/Vc>
```

Comment:
The code needs to compile both with VecCore+Vc `ON` and `OFF`. We made this comment [a while ago](https://github.com/root-project/root/pull/2236#issuecomment-400720614). If you include headers without checking for it like this, the compilation will definitely fail.

Reply:
Yes, the `#ifdef R__HAS_VECCORE` needs to be present here. The latest commit is not supposed to be tested. I updated the PR heading.

Was just updating this PR with more functions till the `Vc` issue gets resolved on my local machine.